pandas/core/internals/managers.py

```diff
@@ -1137,6 +1137,24 @@ def iget_values(self, i: int) -> ArrayLike:
         values = block.iget(self.blklocs[i])
         return values
 
+    @property
+    def column_arrays(self) -> list[np.ndarray]:
+        """
+        Used in the JSON C code to access column arrays.
+        This optimizes compared to using `iget_values` by converting each
+        block.values to a np.ndarray only once up front
+        """
+        arrays = [np.asarray(blk.values) for blk in self.blocks]
```

Comment:
i think this will change the behavior for dt64tz

Reply:
This will not change behaviour, as it is the same as `get_block_values_for_json` did before until a few hours ago. The comment you added to the changed implementation in https://github.com/pandas-dev/pandas/pull/41082 is "Not necessary to override, but helps perf", so assuming that that didn't change behaviour. 

I would also assume it is covered by the tests, but checking explicitly:

Master:
```
In [1]: pd.DataFrame({'a': [1, 2], 'b': pd.date_range("2012", periods=2, tz="Europe/Brussels")}).to_json()
Out[1]: '{"a":{"0":1,"1":2},"b":{"0":1325372400000,"1":1325458800000}}'

In [2]: pd.DataFrame({'b': pd.date_range("2012", periods=2, tz="Europe/Brussels")}).to_json()
Out[2]: '{"b":{"0":1325372400000,"1":1325458800000}}'
```

PR:

```
In [1]: pd.DataFrame({'a': [1, 2], 'b': pd.date_range("2012", periods=2, tz="Europe/Brussels")}).to_json()
Out[1]: '{"a":{"0":1,"1":2},"b":{"0":1325372400000,"1":1325458800000}}'

In [2]: pd.DataFrame({'b': pd.date_range("2012", periods=2, tz="Europe/Brussels")}).to_json()
Out[2]: '{"b":{"0":1325372400000,"1":1325458800000}}'
```

Now, to preserve the performance improvement of https://github.com/pandas-dev/pandas/pull/41082, I can add a check for datetimetz.

Reply:
hopefully more accurate statement: i think this will entail an object-dtype conversion for dt64tz.  I'm not especially bothered by this bc i think this is a good cleanup

Reply:
... which i now see you already addressed in a new commit.  carry on